x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/common_attributes.gen.ts

```diff
@@ -200,7 +200,7 @@ export type AlertsIndexNamespace = z.infer<typeof AlertsIndexNamespace>;
 export const AlertsIndexNamespace = z.string();
 
 export type MaxSignals = z.infer<typeof MaxSignals>;
-export const MaxSignals = z.number().int().min(1);
+export const MaxSignals = z.number().int().min(1).max(1000);
```

Comment:
This will need to be coordinated with a detection-rules release as some [default to 10k](https://github.com/elastic/detection-rules/blob/integration-v8.13.5/rules/integrations/endpoint/elastic_endpoint_security.toml#L21) as well, right?

Or do the pre-built rules flow through a different part of the API?

I'm sure you're tracking this, so don't mind me....just saw these errors with my rules this morning so that's how If found myself here 😅

Reply:
This is a great question, I didn't know we had detection rules with that high of a `max_signals` count. That could potentially cause problems when trying to duplicate rules.

Reply:
Thank you @spong for this catch 🙏 

Additionally, the way it's implemented is a breaking change in the API. If the value is higher than 1000, attempts to create or update a rule will fail with 400, which wasn't the case before. There shouldn't be a max limit at the schema level, we should calculate the final value of max_signals to use in rule executors.

Reply:
> This is a great question, I didn't know we had detection rules with that high of a max_signals count. That could potentially cause problems when trying to duplicate rules.

Yep, I think rule installation and upgrade will be broken as well for those prebuilt rules that have this value higher than 1000.

Reply:
It looks like there are [18 rules](https://github.com/search?q=repo%3Aelastic%2Fdetection-rules%20max_signals%20%3D%2010000&type=code) in the detection-rules repo that have max_signals set to 10k.

Do I get it right that even if we don't merge this PR, in the default Kibana configuration where `xpack.alerting.rules.run.alerts.max == 1000` all these rules, including the Endpoint Security one, will be executing with warnings?

Reply:
Yes, that would be the expected action for rule executions with `max_signals` > 1000 now so those 18 rules would run with warnings out of the box

Reply:
Thank you! Not a strong opinion, but I think this UX might not be the most optimal. If the expectation from the TRaDE team is that those 18 rules should promote as many external alerts as possible (hense the 10k setting), I would suggest to:

1. Either override the system's max limit with the rule's max_signals, for these rules. This could be done via an additional rule parameter `max_signals_force: boolean` and an additional checkbox in the UI. (not sure that it's a good idea).
2. Or not generate a Warning status if `max_signals > xpack.alerting.rules.run.alerts.max` AND it's a prebuilt rule. (looks like a good tradeoff to me).
3. Or maybe set the `max_signals` value for those 18 rules to 1000, if the TRaDE team would be ok with that. (might be not an option if the goal is to promote as many external alerts as possible and let users increase the `xpack.alerting.rules.run.alerts.max` setting for that).

Maybe there's something else we could do. But I think we should try to make sure rules work out of the box with default settings. The Endpoint Security rule is the most basic one, it is installed and enabled automatically with Elastic Defend, and it would be weird to see it not succeeding, unless the user changes some system setting in the Kibana config, for all Kibana instances in their cluster.

@jpdjere @dplumlee Could we please sync on this in the simplified protections channel with the TRADE team, Kseniia and Alex and see what they have to say?

Reply:
I think options 2 or 3 would be my preferences off the top of my head. Something to consider is that even if we don't display a warning, the rule would still be hard capped by the `xpack.alerting.rules.run.alerts.max` config value out of the box and, should the rule ever reach that value (however unlikely), would throw an error from the alerting side of things letting us know we hit it. With that in mind, maybe option 3 would be best, but we'd need to check with TRaDE team for the intent behind that 10k setting.

Reply:
Doesn't look like option 3 is viable based on Justin's response:

> As an example if an environment had say 100k hosts and an upstream integration, such as elastic endpoint happens to generate a batch of alerts over that time, we want to allow them all to match
As you can see, the larger the deployment, the greater the likelihood of hitting that capacity.

My vote goes to the 2nd option

Reply:
Update: https://github.com/elastic/kibana/issues/173593#issuecomment-2124417923